GameServerLib/Logic/API/ApiEventManager.cs

```diff
@@ -0,0 +1,156 @@
+﻿using LeagueSandbox.GameServer.Core.Logic;
+using LeagueSandbox.GameServer.Logic.GameObjects;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+/*
+ * Possible Events:
+[OnActivate]
+[OnAddPAR]
+[OnAllowAdd]
+[OnAssist]
+[OnAssistUnit]
+[OnBeingDodged]
+[OnBeingHit]
+[OnBeingSpellHit]
+[OnCollision]
+[OnCollisionTerrain]
+[OnDeactivate]
+[OnDealDamage]
+[OnDeath]
+[OnDodge]
+[OnHeal]
+[OnHitUnit]
+[OnKill]
+[OnKillUnit]
+[OnLaunchAttack]
+[OnLaunchMissile]
+[OnLevelUp]
+[OnLevelUpSpell]
+[OnMiss]
+[OnMissileEnd]
+[OnMissileUpdate]
+[OnMoveEnd]
+[OnMoveFailure]
+[OnMoveSuccess]
+[OnNearbyDeath]
+[OnPreAttack]
+[OnPreDamage]
+[OnPreDealDamage]
+[OnPreMitigationDamage]
+[OnResurrect]
+[OnSpellCast]
+[OnSpellHit]
+[OnTakeDamage]
+[OnUpdateActions]
+[OnUpdateAmmo]
+[OnUpdateStats]
+[OnZombie]
+ */
+
+namespace LeagueSandbox.GameServer.Logic.API
+{
+    public static class ApiEventManager
+    {
+        private static Game _game;
+        private static Logger _logger;
+
+        internal static void SetGame(Game game)
+        {
+            _game = game;
+            _logger = Program.ResolveDependency<Logger>();
+        }
+
+        public static void removeAllListenersForOwner(Object owner)
+        {
+            OnChampionDamageTaken.RemoveListener(owner);
+            OnUpdate.RemoveListener(owner);
+        }
+        
+        public static EventOnUpdate OnUpdate = new EventOnUpdate();
+        public static EventOnChampionDamageTaken OnChampionDamageTaken = new EventOnChampionDamageTaken();
+        public static EventOnUnitDamageTaken OnUnitDamageTaken = new EventOnUnitDamageTaken();
+    }
+
+
+    public class EventOnUpdate
+    {
+        private List<Tuple<Object, Action<float>>> listeners = new List<Tuple<object, Action<float>>>();
```

Comment:
Replace `Object` with `object`, not only here but the other one's too. (This is not required though I don't feel okay seeing it like that xD)